File: backend-dotnet/Controllers/ProjectsController.cs
using backend_dotnet.Data.Dtos;
using backend_dotnet.Services;
using Microsoft.AspNetCore.Mvc;

namespace backend_dotnet.Controllers
{
    [Route("apiProjects")]
    [ApiController]
    public class ProjectsController : ControllerBase
    {
        private readonly ProjectsServices _service;

        public ProjectsController(ProjectsServices service)
        {
            _service = service;
        }

        [HttpGet("projectList")]
        public IActionResult ProjectsList([FromQuery] string? projectId, [FromQuery] DateOnly? createOn)
        {
            try
            {
                List<ProjectsDto> dtoList = _service.GetProjectList(projectId, createOn);
                if (!dtoList.Any()) return Ok("Don't exists projects");
                return Ok(dtoList);
            }
            catch (Exception e)
            {
                return BadRequest(e.Message);
            }
        }

        [HttpGet("{projectId}")]
        public IActionResult GetProject(string projectId)
        {
            try
            {
                ProjectsDto? dto = _service.GetProject(projectId);
                if (dto == null)
                {
                    return Ok("Project " + projectId + " is not exists");
                }
                return Ok(dto);

            }
            catch (Exception e)
            {
                return BadRequest(e.Message);
            }
        }

        [HttpPost("createProject")]
        public IActionResult CreateProject([FromBody] ProjectsDto projectDto)
        {
            try
            {
                bool isExist = _service.CreateProjects(projectDto);
                if (isExist)
                {
                    return Ok("Create Projects Successfully");
                }
                return BadRequest("Create Projects not Successfully");
            }
            catch (Exception e)
            {
                return BadRequest(e.Message);
            }
        }

        [HttpPut("{projectId}")]
        public IActionResult UpdateProject(string projectId, [FromBody] ProjectsDto projectDto)
        {
            try
            {
                bool isExists = _service.UpdateProject(projectId, projectDto.Name, projectDto.Description);
                if (isExists)
                {
                    return Ok("Update Projects Successfully");
                }
                return BadRequest("Update Projects not Successfully");
            }
            catch (Exception e)
            {
                return BadRequest(e.Message);
            }
        }

        [HttpDelete("{projectId}")]
        public IActionResult DeleteProject(String projectId)
        {
            try
            {
                bool isExists = _service.DeletedProject(projectId);
                if (isExists)
                {
                    return Ok("Delete Projects Successfully");
                }
                return BadRequest("Delete Projects not Successfully");
            }
            catch (Exception e)
            {
                return BadRequest(e.Message);
            }
        }
    }
}

File: backend-python/controllers/projects_controller.py
from fastapi import APIRouter, Depends, Body
from sqlalchemy.orm import Session
from database import get_db
from dtos.projects_dto import ProjectDto
from services import projects_service

controller = APIRouter(prefix="/apiProject", tags=["Projects"])

@controller.get("/projectList")
def get_project_list(dto: ProjectDto = Body(default={}), db: Session = Depends(get_db)):
  try:
    result = projects_service.get_projects_list(dto,db)
    if not result:
      return "Don't exists projects"
    return result
  except Exception as ex:
    return str(ex)


@controller.get("/{project_id}")
def get_project(project_id : str, db : Session = Depends(get_db)):
  try:
    result = projects_service.get_project(project_id, db)
    if not result:
      return "Don't exist project " + project_id 
    return result
  except Exception as ex:
    return str(ex)

@controller.post("/createProject")
def create_project(dto : ProjectDto, db : Session = Depends(get_db)):
  try:
    result = projects_service.create_project(dto,db)
    if result is False:
      return "Project already exists or Id or name is empty"
    return "Create Project Seccessfully"
  except Exception as ex:
    return str(ex)

@controller.put("/{project_id}")
def update_project(project_id : str, dto : ProjectDto, db: Session = Depends(get_db)):
  try:
    result = projects_service.update_project(project_id, dto, db)
    if result is False:
      return "Project not exists"
    return "Update Projects Successfully"
  except Exception as ex:
    return str(ex)

@controller.delete("/{project_id}")
def delete_project(project_id : str, db : Session = Depends(get_db)):
  try:
    result = projects_service.delete_project(project_id, db)
    if result is False:
      return "Project not exists"
    return "Delete Project Successfully"
  except Exception as ex:
    return str(ex)